### PR TITLE
Reputation system

### PR DESCRIPTION
**Pull request by Thanawas-Sirilertsathit**
### Description

- After end_date if participant does not check-in. The reputation point will be deducted.
- After check-in then reputation point increases.
- Reputation is shown in profile page.

### Functional Acceptance Criteria

- #182 
- #183 
- #177 
- #175 
- #187 
- #188 
- #178 

### Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation, if applicable.
- [x] My code follows the Coding Guidelines.

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other

### Additional Information

image

There are progress bars show concurrent activities and progress to next level of reputation

### CODE REVIEW

File: activities/models.py
Comment (OmegaOoh) [inline activities/models.py:1]: Docstrings needed to be written in Sphinx (There are multiple files that the docstring isn't Sphinx)

**Review by OmegaOoh — CHANGES_REQUESTED** [review]
Overall it looks good, but I've some concerns. (written in  reviews)

File: activities/tests/test_host.py [hunk 1]
```diff
@@ -11,9 +12,12 @@ class GrantRemoveHostTest(django.test.TestCase):
 
     def setUp(self):
         """Set up the common URL and create an activity."""
+        DAYSDELTA = 7
```
Comment (OmegaOoh) [inline activities/tests/test_host.py:15]: You should have `_` to separate the words

**Review by OmegaOoh — CHANGES_REQUESTED** [review]
Checked the code, didn't run it yet. Here's my review please fix. ( I will run it later )

File: activities/tests/test_host.py (hunk 1 above)
Comment (opxz7148) [reply activities/tests/test_host.py:15]: Fixed

**Review by OmegaOoh — APPROVED** [review]

**Review by TeerapatTrepopsakulsin — APPROVED** [review]
Nice work! _Approve_

Some suggestions:
- participants can leave just before the activity ends (no point deducted), I think it's better to not let them leave after the activity has started

**Pull request merged**